src/asr_got_reimagined/domain/stages/stage_2_decomposition.py

```
@@ -162,84 +162,133 @@ async def execute(
-                RETURN node.id AS dimension_id
+                batch_node_query = """
+                UNWIND $batch_data AS item
+                MERGE (d:Node {id: item.props.id}) SET d += item.props
+                WITH d, item.type_label_value AS typeLabelValue CALL apoc.create.addLabels(d, [typeLabelValue]) YIELD node
+                RETURN node.id AS created_node_id, item.props.label AS created_label, item.original_identifier AS original_identifier
                 """
-                params_node = {"props": node_props_for_neo4j, "type_label": NodeType.DECOMPOSITION_DIMENSION.value}
-                result_node = execute_query(create_dim_node_query, params_node, tx_type='write')
+                # Using item.props.label as created_label since node_props_for_neo4j contains 'label'
+                results_nodes = execute_query(batch_node_query, {"batch_data": batch_dimension_node_data}, tx_type='write')
                 
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)